mne/time_frequency/tfr.py

```diff
@@ -1331,12 +1333,34 @@ def _plot(self, picks=None, baseline=None, mode='mean', tmin=None,
         n_picks = len(tfr.ch_names) if combine is None else 1
 
         # combine picks
-        if combine == 'mean':
-            data = data.mean(axis=0, keepdims=True)
-        elif combine == 'rms':
-            data = np.sqrt((data ** 2).mean(axis=0, keepdims=True))
-        elif combine is not None:
-            raise ValueError('combine must be None, mean or rms.')
+        _validate_type(combine, (None, str, "callable"))
+        if isinstance(combine, str):
+            _check_option("combine", combine, ("mean", "rms"))
+            if combine == 'mean':
+                data = data.mean(axis=0, keepdims=True)
+            elif combine == 'rms':
+                data = np.sqrt((data ** 2).mean(axis=0, keepdims=True))
+        elif combine is not None:  # callable
+            # It must operate on (n_channels, n_freqs, n_times) and return
+            # (n_freqs, n_times). Operates on a copy in-case 'combine' does
+            # some in-place operations.
+            try:
+                data = combine(data.copy())
+            except TypeError:
+                raise RuntimeError(
+                    "A callable 'combine' must operate on a single argument, "
+                    "a numpy array of shape (n_channels, n_freqs, n_times)."
+                )
+            if (
+                not isinstance(data, np.ndarray)
+                or data.shape != tfr.data.shape[1:]
+            ):
+                raise RuntimeError(
+                    "A callable 'combine' must return a numpy array of shape "
+                    "(n_freqs, n_times)."
+                )
```

Comment:
```suggestion
            _validate_type(data, np.ndarray, 'data returned by callable')
            _check_option('data.shape returned by callable',
                          data.shape, (tfr.data.shape[1:],))
```

Reply:
Honestly, I think the shorter error message with the exact shape is more informative, even if it takes 4 additional lines.

Reply:
What I wrote should tell you the wrong shape that was supplied and the correct shape that is needed, did you check? I could be wrong...

Reply:
Ohh you mean the dimension names. Okay

Reply:
Yes, the names. And, IMO, the phrasing from `_check_option` doesn't really fit this context.